Use auto data-binding for saving to the DB

manolo opened the pull request.

manolo committed:
Use auto data binding for saving to the DB

marcushellberg committed:
initialize overview page once login has rendered to make overall experience snappier

marcushellberg commented:
I tried running this locally, but did not get any content in the grid even though the sync indicator stopped. Didn't have time to investigate further yet

marcushellberg committed:
splash screen, defer loading of overview page until login page is done rendering

marcushellberg committed:
defer service worker load

marcushellberg committed:
don't use is="custom-style"

manolo committed:
Only redirect to https in vaadin.

manolo committed:
Use latest charts

manolo committed:
Cache local and remote db names.

- Offline was not working fine when app starts without connection.
- Bind db names with localstorage
- Some refactoring to handle correctly change of db name

samiheikki commented:
This PR fixes some data syncing issues on IOS Safari that I had and also using `polymer serve -H ip`. Please merge and push to wc.vaadin :)
